File: src/main/java/hdzi/editstarters/initializr/StarterInfo.java
package hdzi.editstarters.initializr;

import hdzi.editstarters.dependency.Dependency;
import hdzi.editstarters.dependency.DependencyScope;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

/**
 * Created by taojinhou on 2018/12/21.
 */
@Data
public class StarterInfo implements Dependency {
    private String id;
    private String name;

    private String description;
    private String versionRange;

    // 坐标信息
    private String groupId;
    private String artifactId;
    private DependencyScope scope;
    private String version;
    private final List<InitializrRepository> repositories = new ArrayList<>();
    private InitializrBom bom;
    private boolean exist = false;

    public void addRepository(InitializrRepository repository) {
        if (repository != null) {
            this.repositories.add(repository);
        }
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        StarterInfo that = (StarterInfo) o;
        return Objects.equals(id, that.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }

    @Override
    public String toString() {
        return this.name;
    }
}

File: src/main/kotlin/hdzi/editstarters/springboot/initializr/StarterInfo.kt
package hdzi.editstarters.springboot.initializr

import hdzi.editstarters.springboot.Dependency

/**
 * Created by taojinhou on 2018/12/21.
 */
class StarterInfo : Dependency {
    lateinit var id: String
    lateinit var name: String
    var description: String? = null
    var versionRange: String? = null

    // 坐标信息
    override lateinit var groupId: String
    override lateinit var artifactId: String
    lateinit var scope: DependencyScope
    var version: String? = null
    var repositories = mutableListOf<InitializrRepository>()
    var bom: InitializrBom? = null
    var exist = false

    fun addRepository(repository: InitializrRepository?) {
        if (repository != null) {
            this.repositories.add(repository)
        }
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as StarterInfo

        if (id != other.id) return false

        return true
    }

    override fun hashCode(): Int {
        return id.hashCode()
    }

    override fun toString(): String {
        return name
    }
}
